docs/topics/addons.rst

```diff
@@ -0,0 +1,348 @@
+
+.. method:: update_addons(config, addons)
+
+    This method is called immediately before :meth:`update_settings`, and should
+    be used to enable and configure other *add-ons* only.
+
+    Add-ons that are added to the :class:`~scrapy.addons.AddonManager` during
+    this callback will also have their :meth:`update_addons` method called.
+
+    :param config: Configuration of this add-on
+    :type config: ``dict``
+
+    :param addons: Add-on manager holding all loaded add-ons
+    :type addons: :class:`~scrapy.addons.AddonManager`
+
+Additionally, add-ons may (and should, where appropriate) provide one or more
+attributes that can be used for limited automated detection of possible
+dependency clashes:
+
+.. attribute:: requires
+
+    list of built-in or custom components needed by this add-on, as strings.
+
+.. attribute:: modifies
+
+    list of built-in or custom components whose functionality is affected or
+    replaced by this add-on (a custom HTTP cache should list ``httpcache`` here)
+
+.. attribute:: provides
+
+    list of components provided by this add-on (e.g. ``mongodb`` for an
+    extension that provides generic read/write access to a MongoDB database)
+
+The entries in the :attr:`requires` and :attr:`modifies` attributes can be add-on
+names or components from other add-ons' :attr:`provides` attribute. You can
+specify :pep:`440`-style information about required versions. Examples::
+
+    requires = ['httpcache']
+    requires = ['otheraddon >= 2.0', 'yetanotheraddon']
+
+The Python object or module that is pointed to by an add-on path (e.g. given in
+the ``ADDONS`` setting, or given to
+:meth:`~scrapy.addons.AddonManager.add`) does not necessarily have to be an
+add-on. Instead, it can provide an ``_addon`` attribute. This attribute can be
+either an add-on or another add-on path.
+
+
+Add-on base class
+=================
+
+Scrapy comes with a built-in base class for add-ons which provides some
+convenience functionality:
+
+* basic settings can be exported via :meth:`~scrapy.addons.Addon.export_basics`,
+  configurable via :attr:`~scrapy.addons.Addon.basic_settings`.
+* a single component (e.g. an item pipeline or a downloader middleware) can be
+  inserted into Scrapy's settings via
+  :meth:`~scrapy.addons.Addon.export_component`, configurable via
+  :attr:`~scrapy.addons.Addon.component_type`,
+  :attr:`~scrapy.addons.Addon.component_key`,
+  :attr:`~scrapy.addons.Addon.component`, and the ``order`` key in
+  :attr:`~scrapy.addons.Addon.default_config`.
+* the add-on configuration can be exposed into Scrapy's settings via
+  :meth:`~scrapy.addons.Addon.export_config`, configurable via
+  :attr:`~scrapy.addons.Addon.default_config`,
+  :attr:`~scrapy.addons.Addon.config_mapping`, and
+  :attr:`~scrapy.addons.Addon.settings_prefix`.
+
+By default, the base add-on class will expose the add-on configuration into
+Scrapy's settings namespace, in caps and with the add-on name prepended. It is
```

Comment:
It seems more flexible to do it without addon name prepended.

Reply:
For example: you make a sublcass of an existing addon, which configures a certain extension. Suddenly the configuration is all wrong, because the extension expects old setting names.